Inconsistent test failures on Python3.7

* xclim version: v0.10.8b0
* Python version: 3.7
* Operating System: Linux and macOS

### Description 
<!--Describe what you were trying to get done.
Tell us what happened, what went wrong, and what you expected to happen.-->
There seems to be an error which only sometimes triggers on Travis CI and causes inconsistent build failures. The most recent event showed up in the build jobs of my most recent PR https://travis-ci.org/Ouranosinc/xclim/jobs/575941873

The stderr is as follows:
```
________________________ TestMissingAnyFills.test_month ________________________
self = <tests.test_checks.TestMissingAnyFills object at 0x122c15ad0>
tasmin_series = <function tasmin_series.<locals>._tasmin_series at 0x122e45b00>

    def test_month(self, tasmin_series):

        ts = tasmin_series(np.zeros(36))
        miss = checks.missing_any(ts, freq="YS", month=7)
        np.testing.assert_equal(miss, [False])

        miss = checks.missing_any(ts, freq="YS", month=8)
        np.testing.assert_equal(miss, [True])

        with pytest.raises(ValueError, match=r"No data for selected period."):
            miss = checks.missing_any(ts, freq="YS", month=1)
    
        miss = checks.missing_any(ts, freq="YS", month=[7, 8])
        np.testing.assert_equal(miss, [True])

        ts = tasmin_series(np.zeros(76))
        miss = checks.missing_any(ts, freq="YS", month=[7, 8])
>       np.testing.assert_equal(miss, [False])

E       AssertionError: 
E       Items are not equal:
E        ACTUAL: <xarray.DataArray (time: 1)>
E       array([ True])
E       Coordinates:
E         * time     (time) datetime64[ns] 2000-01-01
E        DESIRED: [False]
```
Again, this is only happening sometimes. Very strange behaviour.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -3,7 +3,7 @@ default_language_version:
 
 repos:
 -   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.2.3
+    rev: v2.4.0
     hooks:
     -   id: trailing-whitespace
         language_version: python3
@@ -14,13 +14,13 @@ repos:
     -   id: debug-statements
         language_version: python3
 -   repo: https://github.com/ambv/black
-    rev: 19.3b0
+    rev: 19.10b0
     hooks:
     -   id: black
         language_version: python3
         args: ['--target-version=py35']
 -   repo: https://github.com/pycqa/flake8
-    rev: 3.7.7
+    rev: 3.7.9
     hooks:
     -   id: flake8
         language_version: python3
@@ -31,12 +31,12 @@ repos:
 #    -   id: autopep8
 #        args: ['--global-config=setup.cfg','--in-place']
 -   repo: https://github.com/asottile/reorder_python_imports
-    rev: v1.5.0
+    rev: v1.8.0
     hooks:
     -   id: reorder-python-imports
         language_version: python3
 -   repo: https://github.com/asottile/pyupgrade
-    rev: v1.19.0
+    rev: v1.25.1
     hooks:
     -   id: pyupgrade
         language_version: python3

--- a/HISTORY.rst
+++ b/HISTORY.rst
@@ -6,6 +6,7 @@ History
 ------------------------
 * Added type hinting to call signatures of many functions for more explicit type-checking
 * Now supporting explicit builds for Windows OS via Travis CI
+* Fix failing test with Python 3.7
 
 
 0.10.x-beta (2019-06-18)

--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ requirements = [
     "scipy>=1.2",
     "numpy>=1.15",
     "pandas>=0.23",
-    "cftime>=1.0.3",
+    "cftime==1.0.3",
     "netCDF4>=1.4",
     "dask[complete]",
     "bottleneck>=1.2.1",

--- a/tests/test_indices.py
+++ b/tests/test_indices.py
@@ -333,10 +333,6 @@ class TestGrowingDegreeDays:
         assert xci.growing_degree_days(da)[0] == 1
 
 
-@pytest.mark.skipif(
-    sys.version_info >= (3, 7),
-    reason="GrowingSeasonLength causes a dask-related SegFault",
-)
 class TestGrowingSeasonLength:
     def test_simple(self, tas_series):
         # test for different growing length
@@ -956,24 +952,3 @@ def cmip3_day_tas():
     )
     yield ds.tas
     ds.close()
-
-
-@pytest.fixture
-def response():
-    """Sample pytest fixture.
-
-    See more at: http://doc.pytest.org/en/latest/fixture.html
-    """
-    # import requests
-    # return requests.get('https://github.com/audreyr/cookiecutter-pypackage')
-
-
-@pytest.mark.skip
-def test_content(response):
-    """Sample pytest test function with the pytest fixture as an argument."""
-    # from bs4 import BeautifulSoup
-    # assert 'GitHub' in BeautifulSoup(response.content).title.string
-
-
-# x = Test_frost_days()
-# print('done')

--- a/tests/test_precip.py
+++ b/tests/test_precip.py
@@ -258,9 +258,6 @@ class TestMaxNDay:
         assert np.isnan(out1.values[0, -1, -1])
 
 
-@pytest.mark.skipif(
-    sys.version_info < (3, 5), reason="too slow to evaluate on python2.7"
-)
 class TestMaxConsecWetDays:
     # TODO: replace by fixture
     nc_file = os.path.join(TESTS_DATA, "NRCANdaily", "nrcan_canada_daily_pr_1990.nc")
@@ -300,9 +297,6 @@ class TestMaxConsecWetDays:
         # assert (np.isnan(wds.values[0, -1, -1]))
 
 
-@pytest.mark.skipif(
-    sys.version_info < (3, 5), reason="too slow to evaluate on python2.7"
-)
 class TestMaxConsecDryDays:
     # TODO: replace by fixture
     nc_file = os.path.join(TESTS_DATA, "NRCANdaily", "nrcan_canada_daily_pr_1990.nc")

--- a/tests/test_streamflow.py
+++ b/tests/test_streamflow.py
@@ -37,10 +37,11 @@ class TestStats:
     def test_missing(self, ndq_series):
         a = ndq_series
         a = ndq_series.where(~((a.time.dt.dayofyear == 5) * (a.time.dt.year == 1902)))
+        assert a.shape == (5000, 2, 3)
         out = streamflow.stats(a, op="max", month=1)
 
-        np.testing.assert_array_equal(out[1].isnull(), False)
-        np.testing.assert_array_equal(out[2].isnull(), True)
+        np.testing.assert_array_equal(out.sel(time="1900").isnull(), False)
+        np.testing.assert_array_equal(out.sel(time="1902").isnull(), True)
 
 
 class TestFit:

--- a/tests/test_temperature.py
+++ b/tests/test_temperature.py
@@ -761,10 +761,6 @@ class TestDailyFreezeThaw:
         assert np.isnan(frzthw.values[0, -1, -1])
 
 
-@pytest.mark.skipif(
-    sys.version_info >= (3, 7),
-    reason="GrowingSeasonLength causes a dask-related SegFault",
-)
 class TestGrowingSeasonLength:
     def test_single_year(self, tas_series):
         a = np.zeros(366) + K2C

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -69,7 +69,6 @@ class TestDailyDownsampler:
             # assert the values of resampler and grouper are the same
             assert np.allclose(x1.values, x2.values)
 
-    @pytest.mark.skip
     def test_365_day(self):
 
         # 365_day calendar

--- a/tox.ini
+++ b/tox.ini
@@ -16,11 +16,11 @@ python =
 [testenv:black]
 basepython = python
 deps =
-  black
   flake8
+  -e git://github.com/psf/black.git@master#egg=black
 commands =
   flake8 xclim tests
-  black --check xclim tests
+  black --check --target-version=py35 xclim tests
 
 [testenv:docs]
 deps =

--- a/xclim/utils.py
+++ b/xclim/utils.py
@@ -671,29 +671,21 @@ class Indicator:
     _nvar = 1
 
     # CF-Convention metadata to be attributed to the output variable. May use tags {<tag>} formatted at runtime.
-    standard_name = (
-        ""
-    )  # The set of permissible standard names is contained in the standard name table.
+    standard_name = ""  # The set of permissible standard names is contained in the standard name table.
     long_name = ""  # Parsed.
     units = ""  # Representative units of the physical quantity.
     cell_methods = ""  # List of blank-separated words of the form "name: method"
-    description = (
-        ""
-    )  # The description is meant to clarify the qualifiers of the fundamental quantities, such as which
+    description = ""  # The description is meant to clarify the qualifiers of the fundamental quantities, such as which
     #   surface a quantity is defined on or what the flux sign conventions are.
 
     # The `pint` unit context. Use 'hydro' to allow conversion from kg m-2 s-1 to mm/day.
     context = "none"
 
     # Additional information that can be used by third party libraries or to describe the file content.
-    title = (
-        ""
-    )  # A succinct description of what is in the dataset. Default parsed from compute.__doc__
+    title = ""  # A succinct description of what is in the dataset. Default parsed from compute.__doc__
     abstract = ""  # Parsed
     keywords = ""  # Comma separated list of keywords
-    references = (
-        ""
-    )  # Published or web-based references that describe the data or methods used to produce it. Parsed.
+    references = ""  # Published or web-based references that describe the data or methods used to produce it. Parsed.
     comment = (
         ""  # Miscellaneous information about the data or methods used to produce it.
     )

